Swamp Sneak/Assets/Scripts/EnemyLineOfSight.cs

```diff
@@ -71,27 +77,36 @@ public class EnemyLineOfSight : MonoBehaviour {
 	public void SetSightDistance(int sd) {
 		sightDistance = sd;
 	}
-	
+
+	//Function to check if player is stealthed and also handle RNG factor
 	bool stealthCheck()
 	{
-
-		//Function to check if player is stealthed and also handle RNG factor
-		if(GameObject.Find("Player").GetComponent<ThirdPersonCharacter>().getStealth() == false)
-		{
-			return true;
+		if (playerGameObject == null) {
+			playerGameObject = GameObject.FindGameObjectWithTag ("Player");
 		}
-		
-		int chance = Random.Range(1,9);
-		print(chance);
-		
-		if (GameObject.Find("Player").GetComponent<Player>().getStealth() < chance)
+
+		// If player is not crouching...
+		if(playerGameObject.GetComponent<ThirdPersonCharacter>().getStealth() == false)
```

Comment:
getStealth() returns an `int`. Although 0 == false, it may be more clear to explicitly check for 0. Helps debug issues later when going over the code.

Reply:
That's weird, it returns a bool for me? Maybe you were looking at the getStealth function for the Player class? They were initially named the same thing, but I just changed the Player.getStealth() function to Player.getStealthStat() for that very reason, to avoid any potential confusion

Reply:
Ah yeah I was, lol. Thanks